one-ui/src/main/java/com/marklogic/hub/oneui/util/SearchHelper.java

```diff
@@ -96,6 +99,12 @@ public StringHandle search(SearchQuery searchQuery) {
 
         }
         catch (MarkLogicServerException e) {
+            // If there are no entityTypes to search, return empty result set
```

Comment:
This seems like something we can check on before trying to do the search, right? 

Also, I'd say we should add a test - but it doesn't look like SearchHelper is actually being tested at all??? I see in SearchManagerTest that it's being mocked. 

Why is SearchHelper separate from SearchManager? SearchManager isn't doing anything other than delegating to SearchHelper.

I recommend trying to write a real test here. This one is easy, because you don't need any data - just pass in a SearchQuery with no entity names, and verify you get back an empty StringHandle.

Reply:
The reason for adding this check in the catch block is we have to make a getModels call for every search request. This is handled by adding empty list to the collection query if the searchQuery.getQuery().getEntityNames() is empty which results in no results.
 If there are no entityTypes, then there are no search options and we get 400 error with a missing options file. So I added this check in case of exception.

Reply:
I will add a comment as you suggested
"if there are no entity models, then we expect an error because no search options will exist"

Reply:
What about a test for this? Can you submit a SearchQuery with no entity names and get this expected exception, and then verify that you get an empty StringHandle back?

Reply:
Addressed in new commits